onedal/ensemble/tests/test_random_forest.py

```diff
@@ -38,13 +38,28 @@ def test_rf_classifier(queue):
     assert_allclose([1], rf.predict([[0, 0, 0, 0]], queue=queue))
 
 
-@pytest.mark.parametrize("queue", get_queues())
-def test_rf_regression(queue):
+@pytest.mark.parametrize("queue", get_queues("cpu"))
+def test_rf_regression_cpu(queue):
+    X, y = make_regression(
+        n_samples=100, n_features=4, n_informative=2, random_state=0, shuffle=False
+    )
+    rf = RandomForestRegressor(max_depth=2, random_state=0).fit(X, y, queue=queue)
+    if daal_check_version((2023, "P", 300)):
+        assert_allclose([-6.97], rf.predict([[0, 0, 0, 0]], queue=queue), atol=1e-2)
+    else:
+        assert_allclose([-6.83], rf.predict([[0, 0, 0, 0]], queue=queue), atol=1e-2)
```

Comment:
you can get device info from the sycl_queue: 
```python
# Insert here your comment why we have different
# check values for different devices and daal versions.
if queue.sycl_device.is_gpu:
    # do your check
else:
    if daal_check_version((2023, "P", 300)):
        # do your check
    else:
        # do your check
```

Please revert your changes with separating and follow branching above.